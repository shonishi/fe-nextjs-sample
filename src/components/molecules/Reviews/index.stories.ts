import type { Meta, StoryObj } from '@storybook/react';
import Reviews from '.';

const meta: Meta<typeof Reviews> = {
  component: Reviews,
};

export default meta;
type Story = StoryObj<typeof Reviews>;

export const Normal: Story = {
  args: {
    reviews: { href: '#', average: 4, totalCount: 117 },
  },
};
