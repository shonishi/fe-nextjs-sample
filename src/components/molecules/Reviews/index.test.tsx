import Reviews from '.';
import renderer from 'react-test-renderer';

describe('Reviews', () => {
  it('指定された値で表示されること', async () => {
    const tree = renderer
      .create(
        <Reviews
          {...{
            reviews: { href: '#', average: 4, totalCount: 117 },
          }}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
