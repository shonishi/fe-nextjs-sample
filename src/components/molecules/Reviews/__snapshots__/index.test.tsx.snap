// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reviews 指定された値で表示されること 1`] = `
<div
  className="mt-6"
>
  <h3
    className="sr-only"
  >
    Reviews
  </h3>
  <div
    className="flex items-center"
  >
    <div
      className="flex items-center"
    >
      <svg
        aria-hidden="true"
        className="text-gray-900 h-5 w-5 flex-shrink-0"
        fill="currentColor"
        viewBox="0 0 20 20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clipRule="evenodd"
          d="M10.868 2.884c-.321-.772-1.415-.772-1.736 0l-1.83 4.401-4.753.381c-.833.067-1.171 1.107-.536 1.651l3.62 3.102-1.106 4.637c-.194.813.691 1.456 1.405 1.02L10 15.591l4.069 2.485c.713.436 1.598-.207 1.404-1.02l-1.106-4.637 3.62-3.102c.635-.544.297-1.584-.536-1.65l-4.752-.382-1.831-4.401z"
          fillRule="evenodd"
        />
      </svg>
      <svg
        aria-hidden="true"
        className="text-gray-900 h-5 w-5 flex-shrink-0"
        fill="currentColor"
        viewBox="0 0 20 20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clipRule="evenodd"
          d="M10.868 2.884c-.321-.772-1.415-.772-1.736 0l-1.83 4.401-4.753.381c-.833.067-1.171 1.107-.536 1.651l3.62 3.102-1.106 4.637c-.194.813.691 1.456 1.405 1.02L10 15.591l4.069 2.485c.713.436 1.598-.207 1.404-1.02l-1.106-4.637 3.62-3.102c.635-.544.297-1.584-.536-1.65l-4.752-.382-1.831-4.401z"
          fillRule="evenodd"
        />
      </svg>
      <svg
        aria-hidden="true"
        className="text-gray-900 h-5 w-5 flex-shrink-0"
        fill="currentColor"
        viewBox="0 0 20 20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clipRule="evenodd"
          d="M10.868 2.884c-.321-.772-1.415-.772-1.736 0l-1.83 4.401-4.753.381c-.833.067-1.171 1.107-.536 1.651l3.62 3.102-1.106 4.637c-.194.813.691 1.456 1.405 1.02L10 15.591l4.069 2.485c.713.436 1.598-.207 1.404-1.02l-1.106-4.637 3.62-3.102c.635-.544.297-1.584-.536-1.65l-4.752-.382-1.831-4.401z"
          fillRule="evenodd"
        />
      </svg>
      <svg
        aria-hidden="true"
        className="text-gray-900 h-5 w-5 flex-shrink-0"
        fill="currentColor"
        viewBox="0 0 20 20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clipRule="evenodd"
          d="M10.868 2.884c-.321-.772-1.415-.772-1.736 0l-1.83 4.401-4.753.381c-.833.067-1.171 1.107-.536 1.651l3.62 3.102-1.106 4.637c-.194.813.691 1.456 1.405 1.02L10 15.591l4.069 2.485c.713.436 1.598-.207 1.404-1.02l-1.106-4.637 3.62-3.102c.635-.544.297-1.584-.536-1.65l-4.752-.382-1.831-4.401z"
          fillRule="evenodd"
        />
      </svg>
      <svg
        aria-hidden="true"
        className="text-gray-200 h-5 w-5 flex-shrink-0"
        fill="currentColor"
        viewBox="0 0 20 20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clipRule="evenodd"
          d="M10.868 2.884c-.321-.772-1.415-.772-1.736 0l-1.83 4.401-4.753.381c-.833.067-1.171 1.107-.536 1.651l3.62 3.102-1.106 4.637c-.194.813.691 1.456 1.405 1.02L10 15.591l4.069 2.485c.713.436 1.598-.207 1.404-1.02l-1.106-4.637 3.62-3.102c.635-.544.297-1.584-.536-1.65l-4.752-.382-1.831-4.401z"
          fillRule="evenodd"
        />
      </svg>
    </div>
    <p
      className="sr-only"
    >
      4
       out of 5 stars
    </p>
    <a
      className="ml-3 text-sm font-medium text-indigo-600 hover:text-indigo-500"
      href="#"
    >
      117
       reviews
    </a>
  </div>
</div>
`;
